cdap-api/src/main/java/co/cask/cdap/api/workflow/AbstractWorkflow.java

```diff
@@ -126,4 +127,26 @@ protected final void addSpark(String spark) {
     Predicate<WorkflowContext> predicate) {
     return configurer.condition(predicate);
   }
+
+  /**
+   * Adds a local dataset instance to the {@link Workflow}.
+   * See {@link co.cask.cdap.api.dataset.DatasetDefinition} for details.
+   *
+   * @param datasetName name of the dataset instance
+   * @param typeName name of the dataset type
+   * @param properties dataset instance properties
+   */
+  protected final void createLocalDataset(String datasetName, String typeName, DatasetProperties properties) {
```

Comment:
Another question, probably more related to the design. If whether local dataset is being used for input/output of MapReduce job (or Spark) can be determined at runtime, then why the user need to declare it at configure time? (pt1 in the user stories in https://wiki.cask.co/display/CE/Workflow+Enhancement+-+3.4)